snmplib/snmpusm.c

```diff
@@ -3802,7 +3841,12 @@ usm_build_probe_pdu(netsnmp_pdu **pdu)
         user->privProtocolLen = OID_LENGTH(usmNoPrivProtocol);
         user->privProtocol =
             snmp_duplicate_objid(usmNoPrivProtocol, user->privProtocolLen);
-        usm_add_user(user);
+        if (session->flags & SNMP_FLAGS_SESSION_USER) {
```

Comment:
Please add a patch at the start of this series that moves user creation out of usm_build_probe_pdu() such that 'user' is passed as an argument to usm_build_probe_pdu(). I think this is a cleaner solution rather than making usm_build_probe_pdu() modify session properties.

Reply:
Done.
Should I combine all into one commit ?

Reply:
Introducing changes in one commit and undoing these in another commit makes patches much harder to review than necessary. Please combine the "fixup" commit with the commit it fixes up.

Reply:
I merged the latest changes to the previous commit.
Sorry for the question. Do you prefer to have all functional changes in the first commit and ABI only changes  left in another commit? (because some of the changes in the ABI commit were also functional)

Reply:
Either way is fine.

The regression tests still fail. Please fix the regression test failures. The regression tests can be run locally as follows:
```
export MODE=regular && ci/build.sh && ci/net-snmp-run-tests
```

Reply:
I fixed all tests except last 6 tests of 'perl' module related to tcp/udp/unixsocket. They are fail for me in master also.

Reply:
It seems like the regression tests now pass?

Please combine the fixes patch into the first patch.

Reply:
Done